skrf/tests/test_network.py

```diff
@@ -274,6 +274,110 @@ def test_connect_no_frequency(self):
         with self.assertRaises(ValueError):
             ntwk1**ntwk2
 
+    def test_connect_complex_ports(self):
+        """ Test that connecting two networks gives the same resulting network
+        (same Z-parameters) with all s_def when using complex port impedances.
+        """
+
+        # Generate random Z-parameters for two networks
+        z1 = npy.random.uniform(1, 100, size=(1, 2, 2)) +\
+                1j*npy.random.uniform(-100, 100, size=(1, 2, 2))
+        z2 = npy.random.uniform(1, 100, size=(1, 2, 2)) +\
+                1j*npy.random.uniform(-100, 100, size=(1, 2, 2))
+
+        # Port impedances
+        z0_1 = npy.random.uniform(1, 100, size=2) +\
+                1j*npy.random.uniform(-100, 100, size=2)
+        z0_2 = npy.random.uniform(1, 100, size=2) +\
+                1j*npy.random.uniform(-100, 100, size=2)
+        z0_3 = npy.random.uniform(1, 100, size=2) +\
+                1j*npy.random.uniform(-100, 100, size=2)
+
+        # Cascade Z-parameters calculated with ABCD parameters
+        net3_z = rf.a2z(rf.z2a(z1) @ rf.z2a(z2))
+
+        for s_def in rf.S_DEFINITIONS:
+            net3_ref = rf.Network(s=[[0,0],[0,0]], f=1, z0=z0_3, s_def=s_def)
+            net3_ref.z = net3_z
+
+            net1 = rf.Network(s=[[0,0],[0,0]], f=1, z0=z0_1, s_def=s_def)
+            net2 = rf.Network(s=[[0,0],[0,0]], f=1, z0=z0_2, s_def=s_def)
+            net1.z = z1
+            net2.z = z2
```

Comment:
you can use the class method `network.from_z` eventually

Reply:
I wrote some code for supporting initializing networks with all supported parameters and not just s. I'll make a separate PR for it later.

Reply:
> I wrote some code for supporting initializing networks with all supported parameters and not just s. I'll make a separate PR for it later.

like `rf.Network(z=[[0,0],[0,0]], f=1, z0=z0_1, s_def=s_def)` ?

Reply:
yes